pkg/services/query/query.go

```diff
@@ -133,10 +133,31 @@ func (s *Service) QueryData(ctx context.Context, user *user.SignedInUser, skipCa
 // handleExpressions handles POST /api/ds/query when there is an expression.
 func (s *Service) handleExpressions(ctx context.Context, user *user.SignedInUser, parsedReq *parsedRequest) (*backend.QueryDataResponse, error) {
 	exprReq := expr.Request{
-		OrgId:   user.OrgID,
 		Queries: []expr.Query{},
 	}
 
+	if user != nil { // for passthrough authentication, SSE does not authenticate
+		exprReq.User = &backend.User{
+			Login: user.Login,
+			Name:  user.Name,
+			Email: user.Email,
+			Role:  string(user.OrgRole),
+		}
+		exprReq.OrgId = user.OrgID
+	}
+
+	if parsedReq.httpRequest != nil && parsedReq.httpRequest.Header != nil {
+		clippedHeaders := make(map[string]string, len(parsedReq.httpRequest.Header))
+		for k, v := range parsedReq.httpRequest.Header {
+			if len(v) == 0 {
+				clippedHeaders[k] = ""
+				continue
+			}
+			clippedHeaders[k] = v[0]
+		}
+		exprReq.Headers = clippedHeaders
+	}
```

Comment:
This code feels wrong to me. We don't forward all HTTP headers to `QueryData` today so why do we need to do it for SSE?